src/System.Linq.Expressions/tests/BinaryOperators/Coalesce/BinaryCoalesceTests.cs

```diff
@@ -989,6 +989,22 @@ private static void VerifyGenericWithSubClassAndNewRestrictionCoalesce<TCn>(TCn
         #endregion
 
         [Fact]
+        public static void BasicCoalesceExpressionTest()
+        {
+            int? i = 0;
+            double? d = 0;
+            var left = Expression.Constant(d, typeof(double?));
+            var right = Expression.Constant(i, typeof(int?));
+            Expression<Func<double?, int?>> conversion = x => (int?)x;
+
+            BinaryExpression exp = Expression.Coalesce(left, right, conversion);
+
+            Assert.Equal(conversion, exp.Conversion);
+            Assert.Equal(exp.Right.Type, exp.Type);
+            Assert.Equal(ExpressionType.Coalesce, exp.NodeType);
+        }
```

Comment:
Since you are taking care of this, could you also compile and run an expression like this and check the results against what plain C# does? 
Compile with interpretation flag and without - just in case there are bugs in the compiler/interpreter. 

Thanks!!